packages/devtools_app/test/shared/span_parser_test.dart

```diff
@@ -192,1272 +40,196 @@ void spanTester({
-        span: spans[104],
-        scopes: ['storage.modifier.dart'],
-        line: 91,
-        column: 1,
-        length: 4,
-      );
-      spanTester(
-        span: spans[105],
-        scopes: ['storage.type.primitive.dart'],
-        line: 92,
-        column: 1,
-        length: 4,
-      );
-      spanTester(
-        span: spans[106],
-        scopes: ['storage.type.primitive.dart'],
-        line: 93,
-        column: 1,
-        length: 3,
-      );
-    });
-
-    test('punctuation', () {
-      final spans = SpanParser.parse(grammar, punctuation);
-      expect(spans.length, 3);
-      spanTester(
-        span: spans[0],
-        scopes: ['punctuation.comma.dart'],
-        line: 1,
-        column: 1,
-        length: 1,
-      );
-      spanTester(
-        span: spans[1],
-        scopes: ['punctuation.terminator.dart'],
-        line: 2,
-        column: 1,
-        length: 1,
-      );
-      spanTester(
-        span: spans[2],
-        scopes: ['punctuation.dot.dart'],
-        line: 3,
-        column: 1,
-        length: 1,
-      );
-    });
-
-    test('handles malformed input', () {
-      final spans = SpanParser.parse(grammar, openCodeBlock);
-      expect(spans.length, 7);
-
-      // Represents the span covering all lines starting with '///'
-      spanTester(
-        span: spans[0],
-        scopes: ['comment.block.documentation.dart'],
-        line: 1,
-        column: 1,
-        length: 91,
-      );
-
-      // Immediately following '```dart'
-      spanTester(
-        span: spans[1],
-        scopes: [
-          'comment.block.documentation.dart',
-          'variable.other.source.dart',
-        ],
-        line: 2,
-        column: 12,
-        length: 1,
-      );
-
-      // Whitespace after 3rd '///' line.
-      spanTester(
-        span: spans[2],
-        scopes: [
-          'comment.block.documentation.dart',
-          'variable.other.source.dart',
-        ],
-        line: 3,
-        column: 4,
-        length: 1,
-      );
-
-      // 'This should not cause parsing to fail.'
-      spanTester(
-        span: spans[3],
-        scopes: [
-          'comment.block.documentation.dart',
-          'variable.other.source.dart',
-        ],
-        line: 4,
-        column: 4,
-        length: 40,
-      );
-
-      // void
-      spanTester(
-        span: spans[4],
-        scopes: [
-          'storage.type.primitive.dart',
-        ],
-        line: 6,
-        column: 1,
-        length: 4,
-      );
-
-      // main
-      spanTester(
-        span: spans[5],
-        scopes: [
-          'entity.name.function.dart',
-        ],
-        line: 6,
-        column: 6,
-        length: 4,
-      );
-
-      // '// But scope should end when parent scope is closed.'
-      spanTester(
-        span: spans[6],
-        scopes: [
-          'comment.line.double-slash.dart',
-        ],
-        line: 7,
-        column: 3,
-        length: 52,
-      );
+    group('golden', () {
```

Comment:
Just want to double check - these goldens are not platform specific, right? For our image goldens, we only test them on MacOs, but since these are text goldens, this shouldn't matter.

second question, can we use the `equalsGoldenIgnoringHashCodes` helper here? This is what the inspector uses for its text goldens.

Reply:
>  these goldens are not platform specific, right?

Correct. I'm normalizing the newlines when comparing to avoid any failures from different line endings (some Windows users check files out as-is on Windows, meaning the goldens would have `\n` and some have them translated to `\r\n`).

> can we use the `equalsGoldenIgnoringHashCodes` helper here?

It looks like that helper is mainly designed for regexing out any 5-digit hex values following `#` which doesn't apply here. It does also handle adding a suffix for the goldens folder, but I deliberately didn't put these inside `goldens` because everything in there seems to be Flutter-version specific (so we have `goldens` and `goldens_master`), but this is just internal code and not affected by Flutter version at all. We could use it, but we might end up having two copies of the same goldens to update. WDYT?

Reply:
the `goldens_stable` folder is obsolete and something we should delete. For all the other goldens in DevTools, we only test against the version of flutter we pin our bots to (the latest beta). What do you think about putting the syntax_highlighing goldens under `goldens/syntax_highlighting/` so that all the goldens for DevTools are in a consistent place?

Reply:
> the `goldens_stable` folder is obsolete and something we should delete.

Ah, it's still set up in update_goldens by checking the Flutter version. Should that be gone too?

> What do you think about putting the syntax_highlighing goldens under `goldens/syntax_highlighting/` so that all the goldens for DevTools are in a consistent place?

If we're not maintaining two copies of goldens that seems reasonable to me (actually where I put them originally, then saw the suffix stuff). Although, it is nice to have the golden alongside the source file it goes with - but that might not make so much sense if inside `/goldens/`. WDYT?

Reply:
> Ah, it's still set up in update_goldens by checking the Flutter version. Should that be gone too?

Yes that logic should be changed. We should just make sure the the current flutter version matches the flutter version we pin to (flutter-version.txt) in the update_goldens script. We perform this check at the beginning of the build_release.sh script, so we should be able to copy from there, but this can be done in a separate PR.

I think it's okay to have the test data source files under `test_data/` and the goldens under `goldens/`. This would be consistent with most other tests in DevTools.

Reply:
> Yes that logic should be changed. We should just make sure the the current flutter version matches the flutter version we pin to (flutter-version.txt) in the update_goldens script

Ah yes, there's already a TODO explaining that, which I apparently missed.

> I think it's okay to have the test data source files under `test_data/` and the goldens under `goldens/`. This would be consistent with most other tests in DevTools.

sgtm, done!